asserts/database.go

```diff
@@ -89,6 +89,18 @@ var (
 	ErrNotFound = errors.New("assertion not found")
 )
 
+// SupersededRevisionError is the error returned when an assertion is added
```

Comment:
This error type can be made more useful if we cook it with a wider scope:
```
// SupersededRevisionError is observed when an operation attempts to deal with an assertion
// whose revision is no longer current.
type SupersededRevisionError struct {
        superseded, current int
}

func (e *ObsoleteRevisionError) Error() string {
        return fmt.Sprintf("cannot handle assertion with superseded revision %d, current is %d", e.superseded, e.current)
}
```

I'd suggest making both fields private if possible, as suggested above, as that gives us room for tweaking the way we use it without breaking clients. For example, we might use this in cases we don't actually have all the details, or don't want to leak all the details.